NUModules/NUModule.j

```
@@ -3339,7 +3417,16 @@ NUModuleTabViewModeIcon                = 2;
     _categories = someCategories;
     [self didChangeValueForKey:@"categories"];
 
-    _usesPagination = ![_categories count];
+    // Use pagination only if all categories specify a context
+    _usesPagination = YES;
+    for (var i = 0; i < [_categories count]; i++)
+    {
+        if (![_categories[i] contextIdentifier])
+        {
+            _usesPagination = NO;
+            break;
+        }
+    }
```

I think we should force the category to have a fetcher, no ?
If we agree, let's throw an error to ensure all categories have a fetcher and we will fix our current code

---

If we go for it we'll need to change all the controllers with categories, isn't that a big change?

---

I found 56 results, which is not a big deal. It will be part of the PR we will suggest on our repository